src/mantine-hooks/src/use-media-query/use-media-query.ts

```diff
@@ -16,15 +16,25 @@ function attachMediaListener(query: MediaQueryList, callback: MediaQueryCallback
   }
 }
 
-function getInitialValue(query: string) {
+function getInitialValue(query: string, ssrInitialValue?: boolean) {
+  if (ssrInitialValue !== undefined) {
+    return ssrInitialValue;
+  }
+
   if (typeof window !== 'undefined' && 'matchMedia' in window) {
     return window.matchMedia(query).matches;
   }
+
+  // eslint-disable-next-line no-console
+  console.error(
+    '[@mantine/hooks] use-media-query: Please provide a default value when using server side rendering to prevent a hydration mismatch.'
+  );
+
   return false;
 }
 
-export function useMediaQuery(query: string) {
-  const [matches, setMatches] = useState(getInitialValue(query));
+export function useMediaQuery(query: string, ssrInitialValue?: boolean) {
```

Comment:
Let's rename `ssrInitialValue` to `initialValue`

Reply:
Fair enough, I renamed these.